cli/test/README.md

```diff
@@ -0,0 +1,273 @@
+# Cypress CLI Test Suite
+
+This directory contains the test suite for the Cypress CLI package. The tests ensure that the command-line interface behaves correctly across different environments and scenarios.
+
+## Test Structure
+
+### Directory Layout
+
+```
+test/
+├── README.md                    # This file
+├── lib/                        
+│   ├── tasks/
+│   │   ├── verify.spec.ts      # Tests for binary verification
+│   │   ├── install.spec.ts     # Tests for installation process
+│   │   └── ...
+│   ├── util.spec.ts            # Utility function tests
+│   └── ...
+└── fixtures/                   # Test data and mock files
+```
+
+### Key Test Files
+
+#### `lib/tasks/verify.spec.ts`
+**Purpose**: Tests the binary verification process that ensures a downloaded Cypress binary is valid and executable.
+
+**What it covers**:
+- **Given**: A Cypress binary installation (various states: missing, corrupted, valid)
+- **When**: User runs `cypress verify` or verification happens automatically 
+- **Then**: Should correctly identify binary status and provide helpful error messages
+
+**Key scenarios tested**:
+- ✅ Binary exists and is valid → verification succeeds
+- ❌ Binary missing → shows installation instructions
+- ❌ Binary corrupted → provides troubleshooting steps
+- ❌ Binary not executable → suggests permission fixes
+- 🖥️ Platform-specific behavior (Windows, macOS, Linux)
+- 🔄 Retry logic and timeout handling
+
+**Mock Infrastructure**: Uses `createfs()` helper to simulate different file system states without requiring actual binary installations.
+
+## Running Tests
+
+### Prerequisites
+
+Ensure you have Node.js and Yarn installed:
+- Node.js: ^20.1.0 || ^22.0.0 || >=24.0.0
+- Yarn: Latest stable
```

Comment:
This is incorrect. See https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#requirements for Node.js and Yarn.

```shell
yarn set version stable
```

will install the wrong version: Yarn Modern v4.x

Yarn v1 Classic is needed.